oshmem/runtime/oshmem_shmem_init.c

```diff
@@ -146,7 +146,7 @@ int oshmem_shmem_init(int argc, char **argv, int requested, int *provided)
     int ret = OSHMEM_SUCCESS;
 
     if (!oshmem_shmem_initialized) {
-        if (!ompi_mpi_initialized && !ompi_mpi_finalized) {
+        if (ompi_mpi_state < OMPI_MPI_STATE_INIT_COMPLETED) {
```

Comment:
Why not comparing to a not-initialized state here?
If initialization has started already it supposed to be started in some other thread. And doing ompi_init here will launch a parallel init which is not desired.
I'd expect to have a while look polling on a state (with proper memory barriers / atomics) to wait for the started init to complete.

Reply:
The new test is simply doing what the old test did (because `ompi_mpi_initialized` was not set until the end of `ompi_mpi_init()`).  I can change it to be `== OMPI_MPI_STATE_INIT_NOT_INITIALIZED`, since the new variable offers finer-grained control.

If you want it to check for `_STATE_INIT_STARTED` and have it spin for a while, that should probably be a separate PR.